src/Entity/BaseGroup.php

```diff
@@ -0,0 +1,253 @@
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function getSlug(): string
+    {
+        return $this->slug;
+    }
+
+    public function getPostAddress(): PostAddress
+    {
+        return $this->postAddress;
+    }
+
+    public function getDescription(): string
+    {
+        return $this->description;
+    }
+
+    public function isWaitingForApproval(): bool
+    {
+        return self::PENDING === $this->status && !$this->approvedAt;
+    }
+
+    /**
+     * @Algolia\IndexIf
+     */
+    public function isApproved(): bool
+    {
+        return self::APPROVED === $this->status && $this->approvedAt;
+    }
+
+    public function isPending(): bool
+    {
+        return self::PENDING === $this->status;
+    }
+
+    public function isRefused(): bool
+    {
+        return self::REFUSED === $this->status;
+    }
+
+    public function getMembersCount(): int
+    {
+        return $this->membersCounts;
+    }
+
+    public function incrementMembersCount(int $increment = 1)
+    {
+        $this->membersCounts += $increment;
+    }
+
+    public function decrementMembersCount(int $increment = 1)
+    {
+        $this->membersCounts -= $increment;
+    }
+
+    /**
+     * Marks this group as refused/rejected.
+     *
+     * @param string $timestamp
+     */
+    public function refused(string $timestamp = 'now')
+    {
+        $this->status = self::REFUSED;
+        $this->refusedAt = new \DateTime($timestamp);
+        $this->approvedAt = null;
+    }
+
+    public function setName(string $name)
+    {
+        $this->name = $name;
+        $this->canonicalName = static::canonicalize($name);
+    }
+
+    public static function canonicalize(string $name): string
+    {
+        return mb_strtolower($name);
+    }
+
+    public function getCreatedBy(): ?string
+    {
+        return $this->createdBy->toString();
+    }
+
+    public function isCreatedBy(UuidInterface $uuid): bool
+    {
+        return $this->createdBy && $this->createdBy->equals($uuid);
+    }
+
+    public function setPhone(PhoneNumber $phone = null): void
+    {
+        $this->phone = $phone;
+    }
+
+    public function getPhone(): ?PhoneNumber
+    {
+        return $this->phone;
+    }
+
+    /**
+     * Returns the approval date and time.
+     *
+     * @return \DateTime|null
+     */
+    public function getApprovedAt(): ?\DateTime
+    {
+        return $this->approvedAt;
+    }
+
+    public function updateSlug(string $slug)
+    {
+        $this->slug = $slug;
+    }
+
+    public function equals(self $other): bool
+    {
+        return $this->uuid->equals($other->getUuid());
+    }
+
+    public function update(string $name, string $description, PostAddress $address, PhoneNumber $phone)
+    {
+        $this->setName($name);
+        $this->description = $description;
+
+        if (!$this->postAddress->equals($address)) {
```

Comment:
`equals` might break if `$this->postAddress` is `null`. Please consider adding a constructor in this class.